### PR TITLE
chore: compile without nosql's support for Postgres and MySQL

### PR DESCRIPTION
**Pull request by dunglas**
See https://github.com/caddyserver/caddy/pull/6655

### CODE REVIEW

**Comment by dunglas** [comment]
Thanks Francis, sorry I forgot about this PR.

**Comment by steffenbusch** [comment]
My build script, which installs `xcaddy@latest` is just running into issues since v0.4.3 release and the caddy build fails with this error:
```
2024/11/05 22:55:34 [INFO] exec (timeout=0s): /usr/local/go/bin/go build -o /root/out/caddy_2024-11-05_225334 -ldflags -w -s -trimpath -tags nobadger nomysql nopgx
package nomysql is not in std (/usr/local/go/src/nomysql)
package nopgx is not in std (/usr/local/go/src/nopgx)
```

With `go install github.com/caddyserver/xcaddy/cmd/xcaddy@v0.4.2` it still works.

File: builder.go [hunk 1]
```diff
@@ -152,7 +152,7 @@ func (b Builder) Build(ctx context.Context, outputFile string) error {
 			cmd.Args = append(cmd.Args,
 				"-ldflags", "-w -s", // trim debug symbols
 				"-trimpath",
-				"-tags", "nobadger",
+				"-tags", "nobadger", "nomysql", "nopgx",
```
Comment (mohammed90) [inline builder.go:155]: This line should be something like this:
```
"-tags", "nobadger,nomysql,nopgx",
```

**Comment by mohammed90** [comment]
> My build script, which installs `xcaddy@latest` is just running into issues since v0.4.3 release and the caddy build fails with this error:
> 
> ```
> 2024/11/05 22:55:34 [INFO] exec (timeout=0s): /usr/local/go/bin/go build -o /root/out/caddy_2024-11-05_225334 -ldflags -w -s -trimpath -tags nobadger nomysql nopgx
> package nomysql is not in std (/usr/local/go/src/nomysql)
> package nopgx is not in std (/usr/local/go/src/nopgx)
> ```
> 
> With `go install github.com/caddyserver/xcaddy/cmd/xcaddy@v0.4.2` it still works.

Fix in: #213

**Pull request merged**